src/ring_theory/fractional_ideal.lean

```diff
@@ -979,45 +971,37 @@ begin
   rw [eq_span_singleton_of_principal I, generator_I_eq_zero, span_singleton_zero]
 end
 
-lemma one_div_span_singleton (x : g.codomain) :
-  1 / span_singleton x = span_singleton (x⁻¹) :=
-if h : x = 0 then by simp [h] else (right_inverse_eq _ _ (by simp [h])).symm
+-- lemma le_div_iff_of_nonzero {I J J' : fractional_ideal g} (hJ' : J' ≠ 0) :
+--   I ≤ J / J' ↔ ∀ (x ∈ I) (y ∈ J'), x * y ∈ J :=
 
-@[simp]
-lemma span_singleton_inv (x : g.codomain) :
-  (span_singleton x)⁻¹ = span_singleton (x⁻¹) :=
-one_div_span_singleton x
-
-lemma invertible_of_principal (I : fractional_ideal g)
-  [submodule.is_principal (I : submodule R₁ g.codomain)] (h : I ≠ 0) :
-  I * I⁻¹ = 1 :=
-mul_inv_cancel_iff.mpr
-  ⟨span_singleton (generator (I : submodule R₁ g.codomain))⁻¹, mul_generator_self_inv I h⟩
+-- lemma le_div_iff_mul_le {I J J' : fractional_ideal g} (hJ' : J' ≠ 0) : I ≤ J / J' ↔ I * J' ≤ J :=
 
-lemma invertible_iff_generator_nonzero (I : fractional_ideal g)
-  [submodule.is_principal (I : submodule R₁ g.codomain)] :
-  I * I⁻¹ = 1 ↔ generator (I : submodule R₁ g.codomain) ≠ 0 :=
+lemma one_div_span_singleton (x : g.codomain) :
```

Comment:
This proof used to be a one-liner. Why is it now 15+ lines of tactic mode?

Reply:
I am sorry, I do not understand to which `lemma` you are referring to.

Reply:
Oh, now I see. Well, I don't claim that this is the best possible proof, but before the definition of 0^{-1} was already taking care automatically of one part of the proof.

Reply:
I pushed a commit to re-use the previous proof strategy.